programs/cleos/main.cpp

```diff
@@ -2957,6 +2957,97 @@ int main( int argc, char** argv ) {
       std::cout << fc::json::to_pretty_string(unpacked_action_data_json) << std::endl;
    });
 
+   // unpack hex
+   string packed_hex;
+   string type;
+   auto unpack_hex = convert->add_subcommand("unpack_hex", localized("From packed HEX to JSON form"));
+   unpack_hex->add_option("hex", packed_hex, localized("The packed HEX of built-in types including: signed_block, transaction/action_trace, transaction, action, abi_def"))->required();
+   unpack_hex->add_option("--type", type, (localized("Type of the HEX data, if not specified then every type is attempted")));
+   unpack_hex->callback([&] {
+      EOS_ASSERT( packed_hex.size() >= 2, transaction_type_exception, "No packed HEX data found" );
+      vector<char> packed_blob(packed_hex.size()/2);
+      fc::from_hex(packed_hex, packed_blob.data(), packed_blob.size());
+      bool success = false;
+      if (type.empty() || type == "block_state_legacy") {
```

Comment:
If `type` is specified and upacking fails, it should not be attempted on other types. Should just return the failure and return.